vignettes/trial_analysis.Rmd

```diff
@@ -0,0 +1,606 @@
+
+my_stopping <- (StoppingMinCohorts(nCohorts = 3) & 
+                StoppingTargetProb(target = c(0.2, 0.35), prob = 0.5)) |
+               StoppingMinPatients(nPatients = 20)
+
+
+design <- Design(
+  model = model,
+  nextBest = my_next_best,
+  stopping = my_stopping,
+  increments = my_increments,
+  cohortSize = my_size,
+  data = empty_data,
+  startingDose = 3
+)
+```
+
+## Analysing a trial
+Given the trial design constructed above, the process of analysing a real life instance of the trial is simply a matter of providing the model with the actual toxicity status of the participants treated so far.  The escalation rules we defined earlier allow the use of a single patient run-in until either the first DLT is observed or until dose 20 has been administered.  
+
+### The single patient run-in
+Assume that the first three patients - dosed at `1`, `3` and `5` - completed the trial without incident, but that the fourth patient - treated at `10` - experienced a DLT.
+
+We provide this information to `crmPack` via a `Data` object:
+
+```{r}
+firstFour <- Data(
+              x=c(1, 3, 9, 20),
+              y=c(0, 0, 0,  1),
+              ID=1:4,
+              cohort=1:4,
+              doseGrid=doseGrid
+             )
+```
+Within a `Data` object, the doses at which each patient is treated are given by the `x` slot and their toxicity status (a Boolean where a toxicity is represented by a truthy value) by the `y` slot.
+
+The observed data is easily visualised
+```{r}
+plot(firstFour)
+```
+
+and, since the `plot` method returns a `ggplot` object, it is easily customised.
+
+```{r}
+plot(firstFour) + theme_light()
+```
+
+Now, update the model to obtain the posterior estimate of the dose-toxicty curve:
+
+```{r}
+vignetteMcmcOptions <- McmcOptions(burnin=100, step=2, samples=1000)
+postSamples <- mcmc(
+                 data=firstFour, 
+                 model=model, 
+                 options=vignetteMcmcOptions
+               )
+```
+
+The posterior estimate of the dose toxicity curve is easily visualised:
+
+```{r}
+plot(postSamples, model, firstFour)
+```
+
+A visual representation of the model's state is obtained with:
+```{r}
+nextBest(
+  my_next_best, 
+  doselimit=100, 
+  samples=postSamples, 
+  model=model, 
+  data=empty_data
+)$plot
+```
+
+The lower panel of the plot shows the posterior probability that each dose is in the overdose range.  The dashed horizontal black line shows the acceptable risk of overdose: Doses with red lines which go above this line are considered toxic.  The upper panel shows the probability that each dose is in the target toxicity range.  Clearly, a doses of `30` and `45` have the highest probability of being in the target toxicity range.  However, the risk that both are in the overdose range is unacceptable.  Therefore, `20` is the dose recommended for the next cohort.
+
+We can produce a tabulation of the model state with
+```{r}
+tabulatePosterior <- function(mcmcSamples, observedData) {
```

Comment:
if you think this could be generally useful, can you add an issue for adding this to the package? 
then part of that issue would be to replace definition here by mere usage

Reply:
Agree.  I think this - and the graphs and tables that follow - would be useful additions.  I wanted feedback before proceeding.  The functions may better fit in a `crmReporter` package.